pkg/sql/tablemetadatacache/update_table_metadata_cache_job.go

```diff
@@ -152,18 +171,40 @@ func (j *tableMetadataUpdateJobResumer) markAsCompleted(ctx context.Context) {
 
 // updateTableMetadataCache performs a full update of system.table_metadata by collecting
 // metadata from the system.namespace, system.descriptor tables and table span stats RPC.
-func updateTableMetadataCache(ctx context.Context, ie isql.Executor) error {
+func updateTableMetadataCache(
+	ctx context.Context, ie isql.Executor, resumer *tableMetadataUpdateJobResumer,
```

Comment:
If we're passing the entire resumer to this fn can we just make this a method on the job resumer? Or pass the update progress functions to this fn. I think initially I separated this fn to make it easier to test, but it looks like the dependency on the resumer is growing.
We could also just pass the onBatchUpdate callback to this fn, and call the initial updateProgress before calling this fn. Mostly just throwing suggestions out there but I'll leave it up to you.

Reply:
I refactored `tableMetadataUpdater` and moved this logic into a function on that struct. This function has been removed